lib/chess/core/board.rb

```diff
@@ -1,127 +1,119 @@
         @pieces = {
-          WHITE_TEAM => generate_pieces(WHITE_TEAM, 2),
-          BLACK_TEAM => generate_pieces(BLACK_TEAM, 7)
+          WHITE_TEAM => generate_pieces(WHITE_TEAM),
+          BLACK_TEAM => generate_pieces(BLACK_TEAM)
         }
-
-        occup_cells
       end
 
-      def generate_pieces(team, row_of_pawns)
-        main_row = main_row(team, row_of_pawns)
+      private
+
+      def generate_pieces(team)
+        main_index = team == WHITE_TEAM ? 0 : 7
+        pawns_index = team == WHITE_TEAM ? 1 : 6
 
         Pieces.new(
-          team,
-          King.new("e#{main_row}", team),
-          [Queen.new("d#{main_row}", team)],
-          [Bishop.new("c#{main_row}", team), Bishop.new("f#{main_row}", team)],
-          [Knight.new("b#{main_row}", team), Knight.new("g#{main_row}", team)],
-          [Rook.new("a#{main_row}", team, QUEEN_SIDE), Rook.new("h#{main_row}", team, KING_SIDE)],
-          (0..7).to_a.map { |column_index| Pawn.new("#{COLUMNS[column_index]}#{row_of_pawns}", team) }
+          Core::Pieces::King.create_and_occupy(team, @cells[main_index][4]),
+          create_queens(team, main_index),
+          create_bishops(team, main_index),
+          create_knights(team, main_index),
+          create_rooks(team, main_index),
+          create_pawns(team, pawns_index)
         )
       end
 
-      def main_row(team, row_of_pawns)
-        return row_of_pawns + 1 if team == BLACK_TEAM
+      def create_queens(team, main_index)
+        [
+          Core::Pieces::Queen.create_and_occupy(team, @cells[main_index][3])
+        ]
+      end
 
-        row_of_pawns - 1
+      def create_bishops(team, main_index)
+        [
+          Core::Pieces::Bishop.create_and_occupy(team, @cells[main_index][2]),
+          Core::Pieces::Bishop.create_and_occupy(team, @cells[main_index][5])
+        ]
       end
 
-      def occup_cells
-        occup_cells_by(WHITE_TEAM)
-        occup_cells_by(BLACK_TEAM)
+      def create_knights(team, main_index)
+        [
+          Core::Pieces::Knight.create_and_occupy(team, @cells[main_index][1]),
+          Core::Pieces::Knight.create_and_occupy(team, @cells[main_index][6])
+        ]
       end
 
-      def occup_cells_by(team)
-        @pieces[team].to_a.each do |pieces|
-          pieces.each { |piece| find_cell(piece.position.algebraic.to_s).occup_by(piece) }
-        end
+      def create_rooks(team, main_index)
+        [
+          Core::Pieces::Rook.create_and_occupy(team, @cells[main_index][0]),
+          Core::Pieces::Rook.create_and_occupy(team, @cells[main_index][7])
+        ]
       end
 
-      def find_cell(cell_algebraic_string)
-        splitted_cell_string = cell_algebraic_string.split('')
-        @cells[splitted_cell_string[0].to_sym][splitted_cell_string[1].to_i - 1]
+      def create_pawns(team, pawns_index)
+        (0..7).to_a.map do |column_index|
+          Core::Pieces::Pawn.create_and_occupy(team, @cells[pawns_index][column_index])
+        end
```

Comment:
I think there is no need to convert the range to array
```suggestion
        (0..7).map do |column_index|
          Core::Pieces::Pawn.create_and_occupy(team, @cells[pawns_index][column_index])
        end
```

Reply:
that's right